modules/lemmaBidAdapter.js

```diff
@@ -0,0 +1,406 @@
+            adFormat.push({ w: size[0], h: size[1] });
+          }
+        });
+        if (adFormat.length > 0) {
+          bObj.format = adFormat;
+        }
+      }
+    } else {
+      utils.logWarn(LOG_WARN_PREFIX + 'Error: mediaTypes.banner.sizes missing for adunit: ' + bid.params.adunitId);
+    }
+  }
+  return bObj;
+}
+
+function _getVideoRequest(bid) {
+  var vObj;
+  if (cfn(bid.mediaType) === 'video' || utils.deepAccess(bid, 'mediaTypes.video')) {
+    var params = cfn(bid) != '' ? bid.params : null;
+    var sizes = _getSizes(bid);
+    if (cfn(sizes).length == 0) {
+      sizes = cfn(bid.mediaTypes.video.playerSize);
+    }
+    if (cfn(sizes).length > 0) {
+      var videoData = cfn(params.video);
+      vObj = {};
+      if (cfn(videoData) !== '') {
+        vObj = utils.deepClone(videoData);
+      }
+      vObj.w = cfn(sizes[0]);
+      vObj.h = cfn(sizes[1]);
+    } else {
+      utils.logWarn(LOG_WARN_PREFIX + 'Error: mediaTypes.video.sizes missing for adunit: ' + bid.params.adunitId);
+    }
+  }
+  return vObj;
+}
+
+function _getImpressionObject(bid) {
+  var impression = {};
+  var bObj;
+  var vObj;
+  var sizes = bid.hasOwnProperty('sizes') ? bid.sizes : [];
+  var mediaTypes = '';
+  var format = [];
+  var params = cfn(bid.params) != '' ? bid.params : null;
+  impression = {
+    id: cfn(bid.bidId),
+    tagid: cfn(params.adunitId).toString() != '' ? cfn(params.adunitId).toString() : undefined,
+    secure: window.location.protocol === 'https:' ? 1 : 0,
+    bidfloorcur: cfn(params.currency) != '' ? params.currency : DEFAULT_CURRENCY
+  };
+
+  if (cfn(params.bidFloor) != '') {
+    impression.bidfloor = cfn(params.bidFloor);
+  }
+
+  if (bid.hasOwnProperty('mediaTypes')) {
+    for (mediaTypes in bid.mediaTypes) {
+      switch (mediaTypes) {
+        case BANNER:
+          bObj = _getBannerRequest(bid);
+          if (cfn(bObj) !== '') {
+            impression.banner = bObj;
+          }
+          break;
+        case VIDEO:
+          vObj = _getVideoRequest(bid);
+          if (cfn(vObj) !== '') {
+            impression.video = vObj;
+          }
+          break;
+      }
+    }
+  } else {
+    bObj = {
+      pos: 0,
+      w: sizes && sizes[0] ? sizes[0][0] : 0,
+      h: sizes && sizes[0] ? sizes[0][1] : 0,
+    };
+    if (utils.isArray(sizes) && sizes.length > 1) {
+      sizes = sizes.splice(1, sizes.length - 1);
+      sizes.forEach(size => {
+        format.push({
+          w: size[0],
+          h: size[1]
+        });
+      });
+      bObj.format = format;
+    }
+    impression.banner = bObj;
+  }
+
+  return impression.hasOwnProperty(BANNER) ||
+        impression.hasOwnProperty(VIDEO) ? impression : undefined;
+}
+
+function parse(rawResp) {
+  try {
+    if (rawResp) {
+      return JSON.parse(rawResp);
+    }
+  } catch (ex) {
+    utils.logError(LOG_WARN_PREFIX, 'ERROR', ex);
+  }
+  return null;
+}
+
+function cfn(el) {
+  return void 0 === el || el === null ? '' : el;
```

Comment:
Why is the `void` operator needed here? Can this be simplified to `el || ''`?

Also, I noticed `if (cfn(el) != '')` being used throughout the adapter. What is the advantage of using this `cfn` function over `if (el)`?

Reply:
Actually, I have checked undefined and null value in cfn function. I have removed can function and statements.